features/admin.rules.v1/providers/rules-provider.tsx

```diff
@@ -20,104 +20,127 @@ import { SelectChangeEvent } from "@mui/material";
 import React, { ReactNode, useEffect, useState } from "react";
 import { v4 as uuidv4 } from "uuid";
 import RulesContext from "../contexts/rules-context";
+import { ConditionExpressionsMetaDataInterface, RuleExecutionMetaDataInterface } from "../models/meta";
 import {
-    ConditionTypes,
+    AdjoiningOperatorTypes,
+    ConditionExpressionInterface,
+    ConditionExpressionsInterface,
     ExpressionFieldTypes,
-    ExpressionInterface,
-    RuleComponentMetaDataInterface,
-    RuleConditionsInterface,
-    RuleExecutionMetaDataInterface,
+    RuleConditionInterface,
+    RuleExecuteCollectionInterface,
     RuleInterface
 } from "../models/rules";
 
 // Refference to hold the latest context value
-const RuleContextRef: { ruleInstance: RuleInterface[] | undefined } = { ruleInstance: undefined };
+const RuleContextRef: { ruleInstance: RuleExecuteCollectionInterface | undefined } = {
+    ruleInstance: undefined
+};
 
 /**
  * Method to get the context value
  *
  * @returns RuleInstanceData
  */
-export const getRuleInstanceValue = () => RuleContextRef.ruleInstance;
+export const getRuleInstanceValue = () => Object.freeze(RuleContextRef.ruleInstance);
```

Comment:
We'll be moving to to a hook in the future right? Since the object is a nested object, `Object.freeze` wouldn't be ideal.

Lets do a deep clone, if we are planning to keep exposing the instance directly.

```suggestion
export const getRuleInstanceValue = (): RuleExecuteCollectionInterface => Object.freeze(RuleContextRef.ruleInstance);
```